nodes/dummy_node_c/dummy_node_c.go

```diff
@@ -3,20 +3,36 @@ package dummy_node_c
 
 import (
 	"github.com/microsoft/BladeMonRT/nodes"
+	"github.com/microsoft/BladeMonRT/logging"
+	"log"
 )
 
 /** Node that has a hard-coded value for its result. */
 type DummyNodeC struct {
 	nodes.Node
 	result string
+	resultProvider InterfaceResultProvider
 }
 
-func NewDummyNodeC() *DummyNodeC {
-	var dummyNode DummyNodeC = DummyNodeC{}
-	dummyNode.result = "dummy-result-c"
+func NewDummyNodeC(resultProvider InterfaceResultProvider) *DummyNodeC {
```

Comment:
Make a 2nd constructor "forTest" and remove the formal param `resultProvider` here